test/controllers/PrepareUploadControllerSpec.scala

```diff
@@ -22,33 +22,61 @@ class PrepareUploadControllerSpec extends UnitSpec with Matchers with GivenWhenT
 
     val controller = new PrepareUploadController(new StubPrepareUploadService())
 
-    "build and return upload URL if valid request" in {
+    "build and return upload URL if valid request with all data" in {
 
-      Given("there is a valid upload request")
+      Given("there is a valid upload request with all data")
 
       val request: FakeRequest[JsValue] = FakeRequest()
-        .withBody(Json.obj("id" -> "1", "callbackUrl" -> "http://www.example.com"))
+        .withBody(
+          Json.obj(
+            "id"              -> "1",
+            "callbackUrl"     -> "http://www.example.com",
+            "minimumFileSize" -> 0,
+            "maximumFileSize" -> 1024))
 
       When("upload initiation has been requested")
 
       val result = controller.prepareUpload()(request)
 
       Then("service returns valid response with reference and template of upload form")
 
-      status(result) shouldBe 200
+      withClue(result) { status(result) shouldBe 200 }
```

Comment:
if result is a Future wouldn't the message in `withClue` look like `Promise@13aaa14a`